javascript/ql/lib/semmle/javascript/frameworks/Restify.qll

```diff
@@ -157,13 +166,182 @@ module Restify {
     ServerDefinition server;
 
     RouteSetup() {
-      // server.get('/', fun)
-      // server.head('/', fun)
-      server.ref().getAMethodCall(any(Http::RequestMethodName m).toLowerCase()) = this
+      server
+          .ref()
+          .getAMethodCall([
+              "del", "get", "head", "opts", "post", "put", "patch", "param", "pre", "use"
+            ]) = this
     }
 
-    override DataFlow::SourceNode getARouteHandler() { result.flowsTo(this.getArgument(1)) }
+    override DataFlow::SourceNode getARouteHandler() {
+      exists(DataFlow::Node arg |
+        (
+          // server.get('/', fun)
+          // server.get('/', fun1, fun2)
+          // server.get('/', [fun1, fun2])
+          // server.param('name', fun)
+          this.getMethodName() = ["del", "get", "head", "opts", "post", "put", "patch", "param"] and
+          arg = this.getAnArgument() and
+          not arg = this.getArgument(0)
+          or
+          // server.use(fun)
+          // server.use(fun1, fun2)
+          // server.use([fun1, fun2])
+          this.getMethodName() = ["use", "pre"] and
+          arg = this.getAnArgument()
+        ) and
+        (
+          not arg.asExpr() instanceof ArrayExpr and
+          result.flowsTo(arg)
+          or
+          arg.asExpr() instanceof ArrayExpr and
+          result.flowsTo(DataFlow::exprNode(arg.asExpr().(ArrayExpr).getAnElement()))
+        )
+      )
+    }
 
     override DataFlow::Node getServer() { result = server }
   }
+
+  /**
+   * A call to a Restify's createServer method that sets up a formatter.
+   */
+  class FormatterSetup extends DataFlow::CallNode {
+    DataFlow::ObjectLiteralNode formatters;
+
+    FormatterSetup() {
+      // `server = restify.createServer()`
+      this = DataFlow::moduleMember("restify", "createServer").getACall() and
+      exists(DataFlow::ObjectLiteralNode options |
+        options.hasPropertyWrite("formatters", formatters)
+      )
```

Comment:
I think you might be missing an `options = this.getArgument(..)` or similar. 
You are building a cartesian product between all "formatters" and all calls to `createServer`.

Reply:
You are right, thanks! Also added `getALocalSource` approach that you suggested above